### ISSUE TITLE
kube 0.94 release plan

### ISSUE THREAD

**Issue by clux** [kube-rs/kube#1570]
Hoping to send out a release soon. Lots of stuff in the [milestone](https://github.com/kube-rs/kube/milestone/40?closed=1) and the last open PR is ~~basically ready~~merged, so likely will send one out on monday after a doc clean PR (#1573).

## Highlights
Support for `rustls`'s `aws-lc-rs` available under a new `kube/aws-lc-rs` feature.

Two Partial typing improvements;

1. new [`DeserializeGuard`] safety wrapper allows lifting deserialisation errors (for say, not breaking watchers) with `Api::<DeserializeGuard<CaConfigMap>>` type use. See `error_boundary` module  (#1556 and #1558) and new [errorbounded_configmap example](https://github.com/kube-rs/kube/blob/main/examples/errorbounded_configmap_watcher.rs).
2. A derive macro for [`Resource`]; `#[derive(Resource)]` allows inheriting existing `k8s-openapi` resource implementations to avoid stepping into the dynamic api. Via #1565

So basically, better partial typing / safer typing support and rustls, but also some small stuff.
